recipe/build.sh

```diff
@@ -1,7 +1,5 @@
-cp $RECIPE_DIR/uwsgi_config.ini $SRC_DIR
+export UWSGI_PROFILE="${RECIPE_DIR}/uwsgi_config.ini"
+export UWSGI_INCLUDES="${PREFIX}/include,${PREFIX}/include/openssl"
+export LDFLAGS="-L${PREFIX}/lib ${LDFLAGS}"
```

Comment:
Worth a try... Let's see how our build system behaves with that flag.

Reply:
```suggestion
export LDFLAGS="-L${PREFIX}/lib ${LDFLAGS} -Wl, --no-as-needed"
```

Sorry, I forgot to add the "-Wl" flag when writing this by hand, as written it passes "--no-as-needed" to compiler calls. (I wrote this from memory instead of actually referring back to my recipe).

Reply:
And I continue to be an idiot...there's an extra space that sneaked in between the "-Wl," and "--no-as-needed"

Reply:
That finally worked (on Linux at least). However, it appears "--no-as-needed" is only a valid option on GNU ld, so the addition to LDFLAGS need to be made conditional on the non-existence of "MACOSX_DEPLOYMENT_TARGET" Something like this would work I think...

```bash
export LDFLAGS="-L${PREFIX}/lib ${LDFLAGS}"
if [ -z "$MACOXS_DEPLOYMENT_TARGET" ]; then
    export LDFLAGS="${LDFLAGS} -Wl,--no-as-needed"
fi
```
This does seem rather ugly though, so if you can think of a better way to ensure that this is only passed to GNU ld that's probably better. (Maybe simply make a separate build script and use selectors in the meta.yaml?)

I guess more specifically, the issue is that LDFLAGS as predefined by the binutils activation script has "--as-needed" for some reason, so it needs to be disabled, so another option would be to simply add the flag if "--as-needed" is already part of LDFLAGS.
```bash
export LDFLAGS="-L${PREFIX}/lib ${LDFLAGS}"
if [[ ${LDFLAGS} == *"--as-needed"* ]]; then
    export LDFLAGS="${LDFLAGS} -Wl,--no-as-needed"
fi
```